dv/uvm/core_ibex/collect_results.py

```diff
@@ -4,78 +4,217 @@
+
+    if passing_tests:
+        dest.write(box_comment('Details of passing tests'))
+        dest.write('\n')
+        for trr in passing_tests:
+            dest.write(gen_test_run_result_text(trr))
+            dest.write('\n')
+
+    dest.write('\n')
+    dest.write(gen_summary_line(passing_tests, failing_tests))
+    dest.write('\n')
+
+def output_run_results_junit_xml(passing_tests: List[TestRunResult], failing_tests:
+        List[TestRunResult], junit_dest: TextIO, junit_merged_dest: TextIO):
+    '''Write results to JUnit XML
+
+    Two versions are produced a normal version and a merged version. In the
+    normal version there is a test suite per unique test name with a different
+    test case per seed run. In the merged version there is a single test case
+    under the test suite with information for the individual runs merged
+    together. This is to aid use of the Azure Pipelines JUnit dashboard, which
+    doesn't neatly handle the test suite/test case hierarchy
+    '''
+
+    all_tests = passing_tests + failing_tests
 
+    test_suite_cases = {}
+    merged_test_suite_info = {}
+
+    for trr in all_tests:
+        # If we haven't seen this test name yet, create a test suite for it
+        if trr.name not in test_suite_cases:
+            test_suite_cases[trr.name] = []
+            merged_test_suite_info[trr.name] = {
+                    'stdout' : '',
+                    'failures' : ''
+                }
+
+        # Create a test case for the TestRunResult. stdout holds the text
+        # describing the run. Add the same text to failures if the test failed.
+        test_case = junit_xml.TestCase(
+                f'{trr.name}.{trr.seed}')
+
+        test_case.stdout = gen_test_run_result_text(trr)
+
+        merged_test_suite_info[trr.name]['stdout'] += \
+            gen_test_run_result_text(trr)
+
+        if not trr.passed:
+            test_case.add_failure_info(output = trr.failure_message)
+            merged_test_suite_info[trr.name]['failures'] += \
+                    gen_test_run_result_text(trr)
+
+        test_suite_cases[trr.name].append(test_case)
+
+    # Output the normal JUnit XML
+    test_suites = [junit_xml.TestSuite(test_suite_name, test_cases) for
+            test_suite_name, test_cases in test_suite_cases.items()]
+
+    junit_string = junit_xml.to_xml_report_string(test_suites)
+    junit_dest.write(junit_string)
+
+    # Output the merged version of the JUnit XML
+    merged_test_suites = []
+
+    for test_suite_name, test_suite_info in merged_test_suite_info.items():
+        test_case = junit_xml.TestCase(test_suite_name)
+        test_case.stdout = test_suite_info['stdout']
+        test_case.add_failure_info(output = test_suite_info['failures'])
+
+        merged_test_suites.append(junit_xml.TestSuite(test_suite_name,
+            [test_case]))
+
+    junit_merged_string = junit_xml.to_xml_report_string(merged_test_suites)
+    junit_merged_dest.write(junit_merged_string)
```

Comment:
Probably no need for the intermediate variable here:
```suggestion
    junit_merged_dest.write(junit_xml.to_xml_report_string(merged_test_suites))
```